src/plugins/data/public/ui/dataset_selector/advanced_selector.tsx

```diff
@@ -48,22 +56,38 @@ export const AdvancedSelector = ({
         }),
     },
   ]);
-  const [selectedDataset, setSelectedDataset] = useState<BaseDataset | undefined>();
 
-  return selectedDataset ? (
+  const [currentSelectedDataset, setCurrentSelectedDataset] = useState<BaseDataset | undefined>(
+    selectedDataset
+  );
+
+  return currentSelectedDataset ? (
     <Configurator
-      baseDataset={selectedDataset}
+      baseDataset={currentSelectedDataset}
       onConfirm={onSelect}
       onCancel={onCancel}
-      onPrevious={() => setSelectedDataset(undefined)}
+      onPrevious={() => {
+        setSelectedDataset(undefined);
+        setCurrentSelectedDataset(undefined);
+      }}
+      queryService={queryService}
     />
   ) : (
     <DatasetExplorer
       services={services}
       queryString={queryString}
       path={path}
       setPath={setPath}
-      onNext={(dataset) => setSelectedDataset(dataset)}
+      onNext={(dataset) => {
+        setSelectedDataset(dataset);
+        setIndexPattern(dataset.id);
+        setCurrentSelectedDataset(dataset);
+        if (direct) {
```

Comment:
nit: what does "direct" imply? mostly just a comment I don't know what to expect from the argument name